src/idl_gen_fbs.cpp

```diff
@@ -136,8 +236,25 @@ std::string GenerateFBS(const Parser &parser, const std::string &file_name) {
         GenComment(field.doc_comment, &schema, nullptr, "  ");
         schema += "  " + field.name + ":" + GenType(field.value.type);
         if (field.value.constant != "0") schema += " = " + field.value.constant;
-        if (field.IsRequired()) schema += " (required)";
-        if (field.key) schema += " (key)";
+        std::vector<std::string> attributes;
+        if (field.IsRequired()) attributes.push_back("required");
+        if (field.key) attributes.push_back("key");
+
+        if (parser.opts.keep_proto_id) {
+          auto it = proto_fbs_ids.find(field.name);
+          if (it != proto_fbs_ids.end())
```

Comment:
what happens if this fails to find it? We should at least error out instead of silently ignoring it.

Reply:
It can not be failed, maybe I should remove this check